flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -285,9 +285,7 @@ public void run() {
 				if (!jobsToRemove.isEmpty() && processArchiveDeletion) {
 					events.addAll(cleanupExpiredJobs(jobsToRemove));
 				}
-				if (!events.isEmpty()) {
-					updateJobOverview(webOverviewDir, webDir);
-				}
+				updateJobOverview(webOverviewDir, webDir);
```

Comment:
We could also call this once in the constructor; it is supposed to be part of the startup anyway.

Reply:
I'm not really sure. That would yield in empty jobs/overview.json file created on startup, thus the UI would show "no data" icon during startup/job parsing, and then suddenly it would change into list of  jobs once the startup parsing is done.
Up until now the UI showed those "blurred" lines that we understood as "startup in progress" instead.

Reply:
That "blurring" is not really intentional, and the result of the request to the REST API actually returning a 404 and the UI not actually handling it.
From the REST API perspective, this file must be there ASAP. By doing this after the first fetch you introduce some delay, since the HistoryServer still has to query all filesystems first.

We also have to consider cases where there is actually nothing to be done on startup, as jobs are run later on. In this case, showing "no data" is more appropriate, and consistent with the Flink UI.

Reply:
OK, the method `updateJobOverview` is now called from `JobArchiveFetcherTask`'s contructor too